### ISSUE TITLE
Recommend $ and $$ instead of element(by.css()) and element.all(by.css())

### ISSUE THREAD

**Issue by alecxe** [alecxe/eslint-plugin-protractor#2]

**Comment by davidadas** [comment]
Believe it or not, I was actually considering adding that rule as well earlier.

Then I realized that if you were going to do that it css, you might as well start doing it for everything 

e.g.

``` js
$('.foobar'); // vs...
element(by.css('.foobar'));

$('[binding="some.binding"]'); // vs...
element(by.binding('some.binding'));

$('[ng-repeat="x in y"]'); // vs...
element(by.repeater('x in y'));
```

I think it really boils down to a matter of preference at that point, so it might make more sense to write two rules (i.e. one for either approach) and let people pick which one they want to standardize around.

Thoughts?

**Comment by alecxe** [comment]
@davidadas yeah, I had doubts about this rule too. It is definitely not something we should enforce. Even if we do implement it, it has to be 0 (turned off) by default.

As for the "second" rule, are you saying we should warn (allow eslint to warn) if binding/repeat/model.. are used inside CSS selectors suggesting to use the built-in protractor locators instead?

Thanks.

**Comment by davidadas** [comment]
@alecxe for the second one, yes.  Using the CSS selector to query for the bindings, etc, gets messy (much like using XPATHs) so I prefer to have my QA guys use the provided selectors instead:

**Comment by alecxe** [comment]
This issue is now fixed by 335cefaa9da79c6b8739ac94afdc9a11c214f241. Thanks.

Moved the other issue to #5.
